services/slam/fake/data_loader.go

```diff
@@ -113,6 +113,42 @@ func fakeGetInternalState(datasetDir string, slamSvc *SLAM) ([]byte, error) {
 	return data, nil
 }
 
+func fakeGetPointCloudMapStream(datasetDir string, slamSvc *SLAM) (func() ([]byte, error), error) {
+	path := filepath.Clean(artifact.MustPath(fmt.Sprintf(pcdTemplate, datasetDir, slamSvc.getCount())))
+	slamSvc.logger.Debug("Reading " + path)
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	f := func() ([]byte, error) {
+		chunk := make([]byte, 64*1024)
+		bytesRead, err := file.Read(chunk)
+		if err != nil {
+			defer utils.UncheckedErrorFunc(file.Close)
```

Comment:
[minor] we should still `return nil, err` here if one is detected

Reply:
Returning `nil` is equivalent to returning an empty slice. 
I think that returning any data that was written to chunk before an error occurred (current implementation) is better, as I don't see any value in hiding the bytes that were able to be read before an error occurred.

Reply:
That is different behavior than what is in `builtin.go` (helper function in `grpchelper.go`). This should mirror that as exactly as possible, however, I don't feel strongly about where the fix is here to return nil or in the grpchelper function to return the bytes

Reply:
Thanks for the callout. I agree the two should be consistent. I'll change this to be consistent.

Reply:
done